shadows/framework/src/main/java/org/robolectric/shadows/ShadowNoopNativeAllocationRegistry.java

```diff
@@ -1,21 +1,21 @@
 package org.robolectric.shadows;
 
 import static android.os.Build.VERSION_CODES.N;
+import static android.os.Build.VERSION_CODES.P;
 
 import libcore.util.NativeAllocationRegistry;
+import org.robolectric.annotation.ClassName;
 import org.robolectric.annotation.Implementation;
 import org.robolectric.annotation.Implements;
 
 /** Shadow for {@link NativeAllocationRegistry} that is a no-op. */
-@Implements(
-    value = NativeAllocationRegistry.class,
-    minSdk = N,
-    isInAndroidSdk = false,
-    looseSignatures = true)
+@Implements(value = NativeAllocationRegistry.class, minSdk = N, isInAndroidSdk = false)
 public class ShadowNoopNativeAllocationRegistry {
 
-  @Implementation
-  protected Runnable registerNativeAllocation(Object referent, Object allocator) {
+  @Implementation(maxSdk = P)
+  protected Runnable registerNativeAllocation(
```

Comment:
wow, I did not realize there is another code repo for android source code.

https://cs.android.com/android/platform/superproject/+/android-10.0.0_r1:libcore/luni/src/main/java/libcore/util/NativeAllocationRegistry.java

https://cs.android.com/android/platform/superproject/+/android-9.0.0_r61:libcore/luni/src/main/java/libcore/util/NativeAllocationRegistry.java

Looks like function `public Runnable registerNativeAllocation(Object referent, Allocator allocator)` is deleted start from Android10, API 29

Reply:
this function is overloaded, the one still exist has the function signature of 
`Runnable registerNativeAllocation(Object, long)`

Reply:
yes. 

For api <= P, there are actually two `registerNativeAllocation` member functions
https://cs.android.com/android/platform/superproject/+/android-9.0.0_r1:libcore/luni/src/main/java/libcore/util/NativeAllocationRegistry.java;l=171

`public Runnable registerNativeAllocation(Object referent, Allocator allocator)` and `public Runnable registerNativeAllocation(Object referent, long nativePtr)`.

But start from API > P, only `public Runnable registerNativeAllocation(Object referent, long nativePtr)` still exist. https://cs.android.com/android/platform/superproject/+/android-10.0.0_r1:libcore/luni/src/main/java/libcore/util/NativeAllocationRegistry.java;l=230

Reply:
for `public Runnable registerNativeAllocation(Object referent, Allocator allocator)`, I added ClassName and set maxSdk to P; 
for `public Runnable registerNativeAllocation(Object referent, long nativePtr)`, there is no change needed since it exist from N till now, and the first argument is Object type in android source code?

Reply:
emm, for `public Runnable registerNativeAllocation(Object referent, long nativePtr)`, which parameter should I add ClassName annotation?

Reply:
@utzcoz by having a closer look at @Mia0451's PR, and if I understood your comment properly, I think that there is nothing left to do.

The current [ShadowNoopNativeAllocationRegistry](https://github.com/robolectric/robolectric/blob/master/shadows/framework/src/main/java/org/robolectric/shadows/ShadowNoopNativeAllocationRegistry.java) already provides the two `registerNativeAllocation` methods.
The [new version](https://github.com/Mia0451/robolectric/blob/feat/replace_loose_signatures_part40/shadows/framework/src/main/java/org/robolectric/shadows/ShadowNoopNativeAllocationRegistry.java) simply updates the one using loose signature to add `@ClassName` and `maxSdk` to it. The other one does not need to be changed.

In my opinion, this is good to merge.